ADO.NET/OAuthSampleApp/OAuthSampleApp.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Vertica.Data.VerticaClient;
+using Newtonsoft.Json.Linq;
+using System.Configuration;
+
+internal class OAuthSampleApp
+{
+    static void Main(string[] args)
+    {
+        VerticaConnectionStringBuilder vcsb = new VerticaConnectionStringBuilder();
+        vcsb.ConnectionString = ConfigurationManager.ConnectionStrings["ConnectionString"].ConnectionString;
+
+        ConnectToDatabase(vcsb.ConnectionString);
+    }
+
+    public static void ConnectToDatabase(string connectionString)
+    {
+        try
+        {
+            using (VerticaConnection conn = new VerticaConnection(connectionString))
+            {
+                Console.WriteLine("Attempting to connect with connection string: " + ResultSetPrinter.printableConnectionString(connectionString));
+                conn.Open();
+
+                using (VerticaCommand command = conn.CreateCommand())
+                {
+                    Console.WriteLine("Connection Successful");
+                    command.CommandText = ConfigurationManager.AppSettings["Query"];
+                    using (VerticaDataReader reader = command.ExecuteReader())
+                    {
+                        Console.WriteLine("Executing Query: " + command.CommandText);
+                        ResultSetPrinter printer = new ResultSetPrinter(reader);
+                        printer.printResults();
+                        Console.WriteLine("Query Executed. Exiting.");
+                    }
+                }
+            }
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine(ex.Message);
+        }
+    }
+
+    // returns an access token that you can add to the connection string
+    public static async Task<string> GetToken(string tokenType, string clientID, string clientSecret, string user, string password, string tokenUrl)

```

Comment:
How does GetToken get called?  I was expecting code in Main to call it and set the value in the connection string.